src/DataStructures/DenseVector.hpp

```diff
@@ -0,0 +1,43 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines class DenseVector.
+
+#pragma once
+
+#include <blaze/math/DynamicVector.h>
+#include <pup.h>  // IWYU pragma: keep
+
+#include "Utilities/Blaze.hpp"
+
+/*!
+ * \ingroup DataStructuresGroup
+ * \brief A dynamically sized vector of arbitrary type.
+ *
+ * \note This is a thin wrapper around `blaze::DynamicVector`. Please refer to
+ * the [Blaze documentation](https://bitbucket.org/blaze-lib/blaze/wiki/Vectors)
+ * for information on how to use it.
+ */
+template <typename T, bool TF = blaze::defaultTransposeFlag>
+class DenseVector : public blaze::DynamicVector<T, TF> {
+ public:
+  // Inherit constructors
+  using blaze::DynamicVector<T, TF>::DynamicVector;
+
+  /// Charm++ serialization
+  void pup(PUP::er& p) {  // NOLINT
+    auto size = blaze::DynamicVector<T, TF>::size();
+    p | size;
+    if (p.isUnpacking()) {
+      blaze::DynamicVector<T, TF>::resize(size);
+    }
+    if (std::is_fundamental<T>::value) {
```

Comment:
Need `<type_traits>`.

Reply:
use `cpp17::is_fundamental_v` and instead `#include Utilities/TypeTraits.hpp`